### ISSUE TITLE
Allow SRS WKT override for LAS 1.4

### ISSUE THREAD

**Issue by hobu** [PDAL/PDAL#1395]
There are a number of valid WKT definitions that do not transit through GDAL. Especially for 3D GCS situations, it would be useful if `writer.las` for LAS 1.4 had a no-matter-what clobber whatever PDAL thinks the SRS should be and simply output the WKT specified via user option.

**Comment by abellgithub** [comment]
Should we fix GDAL?  Seems better.

**Comment by hobu** [comment]
Yes, but the task there is huge. GDAL, Proj.4, and libgeotiff all need to be updated to support 3D GCS's, for example. This hack would give people an escape valve when all they are trying to do is accurately describe the data.

**Comment by abellgithub** [comment]
I get that.  What happens if the input has one of these weird SRSes?

**Comment by hobu** [comment]
This is a valid 3D GCS that does not transit through GDAL:
```
GEODCRS["NAD83(HARN)",  
    DATUM["NAD83 (High Accuracy Reference Network)",  
    ELLIPSOID["GRS 1980", 6378137, 298.257222101, LENGTHUNIT["metre", 1.0]]],  
    CS[ellipsoidal, 3],  
        AXIS["latitude", north, 
        ORDER[1], ANGLEUNIT["degree", 0.01745329252]],  
        AXIS["longitude", east, ORDER[2], 
            ANGLEUNIT["degree", 0.01745329252]],  
        AXIS["ellipsoidal height", up, 
        ORDER[3], 
        LENGTHUNIT["metre", 1.0]],  
    ID["EPSG", 4957]]

```

**Comment by abellgithub** [comment]
Right, but what do you mean "does not transit"?  What happens?

**Comment by hobu** [comment]
We also have an issue with PDAL's (over) use of OSRValidate. It fails for `EPSG:3857` in GDAL 1.11.x:
```
[hobu@pyro gdal (trunk)]$ gdalsrsinfo -V "EPSG:3857"
gdalsrsinfo: got arg #1 : [-V]
gdalsrsinfo: got arg #2 : [EPSG:3857]
gdalsrsinfo: trying to open with GDAL
ERROR 4: `EPSG:3857' does not exist in the file system,
and is not recognised as a supported dataset name.

gdalsrsinfo: trying to open with OGR
OGR: OGROpen(EPSG:3857) failed.
gdalsrsinfo: did not open with OGR
gdalsrsinfo: trying to get SRS from user input [EPSG:3857]
gdalsrsinfo: got SRS from user input
gdalsrsinfo: bGotSRS: 1 bValidate: 1 pszOutputType: default bPretty: 0
OGRSpatialReference::Validate: Parsing error : syntax error, unexpected ',', expecting ']'. Error occured around:
ts=m +nadgrids=@null +wktext  +no_defs"],AUTHORITY["EPSG","3857"]]
                                        ^

Validate Fails - SRS is not well formed

gdalsrsinfo: PrintSRS( oSRS, proj4, 1, 1 )

PROJ.4 : '+proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +wktext  +no_defs'

gdalsrsinfo: PrintSRS( oSRS, wkt, 1, 1 )

OGC WKT :
PROJCS["WGS 84 / Pseudo-Mercator",
    GEOGCS["WGS 84",
        DATUM["WGS_1984",
            SPHEROID["WGS 84",6378137,298.257223563,
                AUTHORITY["EPSG","7030"]],
            AUTHORITY["EPSG","6326"]],
        PRIMEM["Greenwich",0,
            AUTHORITY["EPSG","8901"]],
        UNIT["degree",0.0174532925199433,
            AUTHORITY["EPSG","9122"]],
        AUTHORITY["EPSG","4326"]],
    PROJECTION["Mercator_1SP"],
    PARAMETER["central_meridian",0],
    PARAMETER["scale_factor",1],
    PARAMETER["false_easting",0],
    PARAMETER["false_northing",0],
    UNIT["metre",1,
        AUTHORITY["EPSG","9001"]],
    AXIS["X",EAST],
    AXIS["Y",NORTH],
    EXTENSION["PROJ4","+proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +wktext  +no_defs"],
    AUTHORITY["EPSG","3857"]]

GDAL: In GDALDestroy - unloading GDAL shared library.
```

**Comment by abellgithub** [comment]
Don't we already have this for `writers.las` with the options `a_srs`?  Are you suggesting that we do this on the reader?

**Comment by hobu** [comment]
No we don't have this. When someone assigns the WKT, it transits through GDAL's WKT interpreter. If there is WKT that GDAL doesn't recognize, it can get stripped off.

**Comment by abellgithub** [comment]
Unless I misread, this shouldn't be the case according to the code.  The SRS that's set into the spatial reference should come out as it was written.  Do you have a test case where this is failing?

**Comment by abellgithub** [comment]
Believe this is fixed with cb3f2ce14332ec9998fec264ff301b0d1de85976.  @hobu, can you verify that this now works with `a_srs`?

**Comment by abellgithub** [comment]
You can now use the `a_srs` option on `writers.las` to set a WKT coordinate system to be written as provided.  43381163027cdcc7306938999e251c645fd2f9aa
